core/actor/src/main/scala/net/liftweb/actor/LAFuture.scala

```diff
@@ -406,5 +408,28 @@ object LAFuture {
 
     ret
   }
-}
 
+  private def inContext[T](f: () => T, context: Box[LAFuture.Context]): () => T = {
+    context.map(_.around(f)) openOr f
+  }
+
+  private def inContext[A, T](f: (A) => T, context: Box[LAFuture.Context]): (A) => T = {
+    context.map(_.around(f)) openOr f
+  }
+
+  /**
+    * Allows to wrap function in another function providing some additional functionality.
+    * It may choose to execute or not execute that functionality, but should not interpret
+    * or change the returned value; instead, it should perform orthogonal actions that
+    * need to occur around the given functionality. Typical example is setting up DB
+    * transaction.
+    *
+    * This is similar to [[net.liftweb.common.CommonLoanWrapper]], however, it decorates the
+    * function eagerly. This way, you can access current thread's state which is essential
+    * to set up e.g. HTTP session wrapper.
```

Comment:
This would read better as `which is essential to do things like set up an HTTP session wrapper`